sympy/solvers/ode.py

```diff
@@ -2410,6 +2410,21 @@ def checkodesol(ode, sol, func=None, order='auto', solve_for_func=True):
     x = func.args[0]
     while s:
         if testnum == 0:
+            lhs, rhs = ode.lhs, ode.rhs
+            temp_l, temp_r = None, None
+            if sol.lhs == func and not sol.rhs.has(func):
+                temp_l, temp_r = sol.rhs, sol.lhs
+            elif sol.rhs == func and not sol.lhs.has(func):
+                temp_l, temp_r = sol.lhs, sol.rhs
+            if temp_l is not None and temp_r is not None:
+                for i in range(order,-1,-1):
+                    lhs = simplify(lhs.subs(func.diff(x, i), temp_l.diff(x, i)))
+                    rhs = simplify(rhs.subs(func.diff(x, i), temp_r.diff(x, i)))
+                if lhs == 0 and rhs == 0:
```

Comment:
What if the lhs and rhs are equal but non-zero? This will not work most of the time.

Reply:
oh I forgot that point.